test/functional/rpc_netinfo.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Dash Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test network information fields across RPCs."""
+
+from test_framework.util import (
+    assert_equal
+)
+from test_framework.script import (
+    hash160
+)
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    MasternodeInfo,
+    p2p_port
+)
+from test_framework.test_node import TestNode
+
+from _decimal import Decimal
+from random import randint
+
+# See CMainParams in src/chainparams.cpp
+DEFAULT_PORT_MAINNET_CORE_P2P = 9999
+# See CRegTestParams in src/chainparams.cpp
+DEFAULT_PORT_PLATFORM_P2P = 22200
+DEFAULT_PORT_PLATFORM_HTTP = 22201
+
+class Node:
+    mn: MasternodeInfo
+    node: TestNode
+    platform_nodeid: str = ""
+
+    def __init__(self, node: TestNode, is_evo: bool):
```

Comment:
@coderabbitai, please investigate if any of these test cases can semi-trivially / easily be converted into unit tests via c++ unit testing, instead of being inside functional tests